Multiple definitions of `struct_copy` function in submodule

`a.f90`:
```fortran
module person_module
    implicit none

    type :: Person
        character(len=30) :: name
        integer :: age
    end type Person

end module person_module
```


`b.f90`:
```fortran
program main
    use person_module
    implicit none

    type(Person) :: p1
    character(len=:), allocatable :: message

    p1%name = "Jinang Shah"
    p1%age  = 25
end program main
```



LFortran Error:
```console
(lf) jinang_shah@JNSLAP:~/Desktop/lfortran$ lfortran --new-classes --separate-compilation stdliba.f90 stdlibc.f90 -o temp
/home/jinang_shah/conda_root/envs/lf/bin/../lib/gcc/x86_64-conda-linux-gnu/14.2.0/../../../../x86_64-conda-linux-gnu/bin/ld: /tmp/stdlibc.tmp_aWqub9FqPOvkL6SLFAG4A9.o: in function `_copy_person_module_person':
LFortran:(.text+0x0): multiple definition of `_copy_person_module_person'; /tmp/stdliba.tmp_aWqub9FqPOvkL6SLFAG4A9.o:LFortran:(.text+0x0): first defined here
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
The command 'clang -o temp /tmp/stdliba.tmp_aWqub9FqPOvkL6SLFAG4A9.o /tmp/stdlibc.tmp_aWqub9FqPOvkL6SLFAG4A9.o  -L"/home/jinang_shah/Desktop/lfortran/src/bin/../runtime" -Wl,-rpath,"/home/jinang_shah/Desktop/lfortran/src/bin/../runtime" -llfortran_runtime -lm' failed.
Tip: If there is a linker issue, switch the linker using --linker=<CC> option or create an environment variable `export LFORTRAN_LINKER=<CC>`, where CC is clang or gcc
Also, if required use --linker-path=<PATH>, where PATH has location to look for the linker execuatable
```

--- a/integration_tests/CMakeLists.txt
+++ b/integration_tests/CMakeLists.txt
@@ -1660,6 +1660,7 @@ RUN(NAME derived_types_80 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc EXTRA_AR
 RUN(NAME derived_types_81 LABELS gfortran llvm)
 RUN(NAME derived_types_82 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc EXTRA_ARGS --realloc-lhs-arrays --new-classes)
 RUN(NAME derived_types_83 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
+RUN(NAME derived_types_84 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc EXTRAFILES derived_types_84_module.f90 EXTRA_ARGS --separate-compilation)
 RUN(NAME derived_type_with_default_init LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 RUN(NAME derived_type_with_default_init_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 

--- a/integration_tests/derived_types_84.f90
+++ b/integration_tests/derived_types_84.f90
@@ -1,0 +1,9 @@
+program derived_types_84
+    use derived_types_84_module
+    implicit none
+    type(Person) :: p1
+    p1%age  = 25
+    call get_age(p1)
+    print *, p1%age
+    if (p1%age /= 30) error stop
+end program derived_types_84

--- a/integration_tests/derived_types_84_module.f90
+++ b/integration_tests/derived_types_84_module.f90
@@ -1,0 +1,17 @@
+module derived_types_84_module
+    implicit none
+
+    type :: Person
+        integer :: age
+    end type Person
+
+contains
+
+subroutine get_age(p)
+    type(Person), intent(inout) :: p
+    p%age = 30
+end subroutine get_age
+
+end module derived_types_84_module
+
+

--- a/src/libasr/codegen/llvm_utils.cpp
+++ b/src/libasr/codegen/llvm_utils.cpp
@@ -8536,7 +8536,7 @@ llvm::Value* LLVMUtils::handle_global_nonallocatable_stringArray(Allocator& al, 
                             + "_" + ASRUtils::symbol_name(struct_sym);
         llvm::Function *func = llvm::Function::Create(
             funcType,
-            llvm::Function::ExternalLinkage,
+            llvm::Function::LinkOnceODRLinkage,
             func_name,
             module
         );

--- a/tests/reference/llvm_new_classes-classes1-bfe507d.json
+++ b/tests/reference/llvm_new_classes-classes1-bfe507d.json
@@ -6,7 +6,7 @@
     "outfile": null,
     "outfile_hash": null,
     "stdout": "llvm_new_classes-classes1-bfe507d.stdout",
-    "stdout_hash": "06728dd4bb5a108a9ce97aa7d9cc56fd4000fc93eaebae8fece9379b",
+    "stdout_hash": "108f99a4f57acddfa20108e7a26be76105950747079bf9fa06df4924",
     "stderr": null,
     "stderr_hash": null,
     "returncode": 0

--- a/tests/reference/llvm_new_classes-classes1-bfe507d.stdout
+++ b/tests/reference/llvm_new_classes-classes1-bfe507d.stdout
@@ -24,7 +24,7 @@ return:                                           ; preds = %.entry
   ret void
 }
 
-define void @_copy_xx_base(i8* %0, i8* %1) {
+define linkonce_odr void @_copy_xx_base(i8* %0, i8* %1) {
 entry:
   %2 = bitcast i8* %0 to %base*
   %3 = bitcast i8* %1 to %base*

--- a/tests/reference/llvm_new_classes-classes2-7de4211.json
+++ b/tests/reference/llvm_new_classes-classes2-7de4211.json
@@ -6,7 +6,7 @@
     "outfile": null,
     "outfile_hash": null,
     "stdout": "llvm_new_classes-classes2-7de4211.stdout",
-    "stdout_hash": "ea5dc0fe85068bd865da42ae7b38c06a3d39de7eb23e231f95525a36",
+    "stdout_hash": "ddd4aee5f962e4e9281afdf31bb70173405b477c966424580b15280a",
     "stderr": null,
     "stderr_hash": null,
     "returncode": 0

--- a/tests/reference/llvm_new_classes-classes2-7de4211.stdout
+++ b/tests/reference/llvm_new_classes-classes2-7de4211.stdout
@@ -33,7 +33,7 @@ return:                                           ; preds = %.entry
   ret float %6
 }
 
-define void @_copy_defs_point(i8* %0, i8* %1) {
+define linkonce_odr void @_copy_defs_point(i8* %0, i8* %1) {
 entry:
   %2 = bitcast i8* %0 to %point*
   %3 = bitcast i8* %1 to %point*
@@ -46,7 +46,7 @@ entry:
   ret void
 }
 
-define void @_copy_defs_point2d(i8* %0, i8* %1) {
+define linkonce_odr void @_copy_defs_point2d(i8* %0, i8* %1) {
 entry:
   %2 = bitcast i8* %0 to %point2d*
   %3 = bitcast i8* %1 to %point2d*

--- a/tests/reference/llvm_new_classes-select_type_13-cafec20.json
+++ b/tests/reference/llvm_new_classes-select_type_13-cafec20.json
@@ -6,7 +6,7 @@
     "outfile": null,
     "outfile_hash": null,
     "stdout": "llvm_new_classes-select_type_13-cafec20.stdout",
-    "stdout_hash": "bae3cb871507d8ec233eb3096d273bcb4c78e359015a091464b72135",
+    "stdout_hash": "dede7ec56899837af8e39e075d8eefa9e8af83957fbdb7807b6c03a5",
     "stderr": null,
     "stderr_hash": null,
     "returncode": 0

--- a/tests/reference/llvm_new_classes-select_type_13-cafec20.stdout
+++ b/tests/reference/llvm_new_classes-select_type_13-cafec20.stdout
@@ -86,7 +86,7 @@ return:                                           ; preds = %.entry
   ret float %5
 }
 
-define void @_copy_select_type_13_module_shape(i8* %0, i8* %1) {
+define linkonce_odr void @_copy_select_type_13_module_shape(i8* %0, i8* %1) {
 entry:
   %2 = bitcast i8* %0 to %shape*
   %3 = bitcast i8* %1 to %shape*
@@ -99,7 +99,7 @@ entry:
   ret void
 }
 
-define void @_copy_select_type_13_module_circle(i8* %0, i8* %1) {
+define linkonce_odr void @_copy_select_type_13_module_circle(i8* %0, i8* %1) {
 entry:
   %2 = bitcast i8* %0 to %circle*
   %3 = bitcast i8* %1 to %circle*
@@ -127,7 +127,7 @@ ifcont:                                           ; preds = %else, %then
   ret void
 }
 
-define void @_copy_select_type_13_module_rectangle(i8* %0, i8* %1) {
+define linkonce_odr void @_copy_select_type_13_module_rectangle(i8* %0, i8* %1) {
 entry:
   %2 = bitcast i8* %0 to %rectangle*
   %3 = bitcast i8* %1 to %rectangle*

